app/models/concerns/externally_identifiable.rb

```diff
@@ -4,6 +4,19 @@ module ExternallyIdentifiable
   extend ActiveSupport::Concern
 
   module ClassMethods
+    def efind(use_id)
+      find_by(id: use_id) || find_by(external_id: use_id)
+    end
+
+    def efind!(use_id, api: false)
+      ret = efind(use_id)
+      return ret if ret.present?
+
+      raise NoSuchRecordError.new(self::API_OBJECT, external_id: true) if api
+
+      raise ActiveRecord::RecordNotFound, "#{name} could not be found with id or external_id: #{use_id}"
```

Comment:
Is `name` defined somewhere else?

Reply:
This is a concern included in a class, so `name` is the name of the class, in this case our models.  So it's something like `Customer could not be found...`